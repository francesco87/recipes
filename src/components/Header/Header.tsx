import { useNavigate } from 'react-router';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

const Header = () => {
  const navigate = useNavigate();

  return (
    <Box
      sx={{
        width: '100%',
        paddingBottom: 4,
      }}
    >
      <Box>
        <Button onClick={() => navigate(-1)} startIcon={<ChevronLeftIcon />}>
          Go results list
        </Button>
      </Box>
    </Box>
  );
};

export default Header;
