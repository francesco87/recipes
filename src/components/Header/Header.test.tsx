import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, vi, expect } from 'vitest';
import Header from './Header';

const mockNavigate = vi.fn();

vi.mock('react-router', () => ({
  useNavigate: () => mockNavigate,
}));

describe('Header Component', () => {
  it('Render Go results list Button', () => {
    render(<Header />);
    const button = screen.getByText('Go results list');
    expect(button).toBeInTheDocument();
  });

  it('Go back when click on button', () => {
    render(<Header />);
    const button = screen.getByText('Go results list');
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });

  it('Header Snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
