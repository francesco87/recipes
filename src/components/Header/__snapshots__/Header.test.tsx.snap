// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > Header Snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-12luabx"
  >
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
        Go results list
      </button>
    </div>
  </div>
</div>
`;
